presto-main/src/main/java/com/facebook/presto/sql/gen/OrderingCompiler.java

```diff
@@ -216,6 +240,101 @@ private static void generatePageIndexCompareTo(ClassDefinition classDefinition,
                 .retInt();
     }
 
+    public PageWithPositionComparator compilePageWithPositionComparator(List<Type> sortTypes, List<Integer> sortChannels, List<SortOrder> sortOrders)
+    {
+        requireNonNull(sortTypes, "sortTypes is null");
+        requireNonNull(sortChannels, "sortChannels is null");
+        requireNonNull(sortOrders, "sortOrders is null");
+
+        try {
+            return pageWithPositionComparators.get(new PagesIndexComparatorCacheKey(sortTypes, sortChannels, sortOrders));
+        }
+        catch (ExecutionException e) {
+            throw new RuntimeException(e.getCause());
+        }
+    }
+
+    private PageWithPositionComparator internalCompilePageWithPositionComparator(List<Type> types, List<Integer> sortChannels, List<SortOrder> sortOrders)
+    {
+        PageWithPositionComparator comparator;
+        try {
+            Class<? extends PageWithPositionComparator> pageWithPositionsComparatorClass = generatePageWithPositionComparatorClass(types, sortChannels, sortOrders);
+            comparator = pageWithPositionsComparatorClass.newInstance();
+        }
+        catch (Throwable t) {
+            log.error(t, "Error compiling merge sort comparator for channels %s with order %s", sortChannels, sortChannels);
```

Comment:
the second `sortChannels` ->  `sortOrders`

Reply:
+1